### PR TITLE
Update sei-chain usages to remove some deferred operations

### PR DESCRIPTION
**Pull request by udpatil**
## Describe your changes and provide context
This refactors sei-chain to remove deferred send usages for this PR: https://github.com/sei-protocol/sei-cosmos/pull/266

## Testing performed to validate your change
local chain testing, dex behavior needs testing / review for sudo endblock behavior.

### CODE REVIEW

File: x/dex/keeper/abci/end_block_deposit.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (w KeeperWrapper) GetDepositSudoMsg(ctx sdk.Context, typedContractAddr type
 	if err != nil {
 		panic(err)
 	}
-	if err := w.BankKeeper.DeferredSendCoinsFromModuleToAccount(ctx, types.ModuleName, contractAddr, escrowed); err != nil {
+	if err := w.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, contractAddr, escrowed); err != nil {
```
Comment (codchen) [inline x/dex/keeper/abci/end_block_deposit.go:49]: this is okay since deposit is handled sequentially

Comment (BrandonWeng) [reply x/dex/keeper/abci/end_block_deposit.go:49]: I think we updated it to be concurrent with the overall dex endblock optimization

Comment (BrandonWeng) [reply x/dex/keeper/abci/end_block_deposit.go:49]: But the optimization is not large, we can make it sequential again if it adds complications

Comment (BrandonWeng) [reply x/dex/keeper/abci/end_block_deposit.go:49]: https://github.com/sei-protocol/sei-chain/blob/52cf5679e95f9e491596b62588a8525b4bcd1647/x/dex/contract/abci.go#L179-L194

**Review by philipsu522 — APPROVED** [review]

**Review by codchen — APPROVED** [review]

**Review by BrandonWeng — APPROVED** [review]

**Pull request merged**